packages/react/src/SelectPanel/SelectPanel.features.stories.tsx

```diff
@@ -422,3 +423,71 @@ export const AsyncFetch: StoryObj<typeof SelectPanel> = {
     },
   },
 }
+
+export const NoItems = () => {
+  const [selected, setSelected] = React.useState<ItemInput[]>([])
+  const [filteredItems, setFilteredItems] = React.useState<ItemInput[]>([])
+  const [open, setOpen] = useState(true)
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const onFilterChange = (value: string = '') => {
+    setTimeout(() => {
+      // fetch the items
+      setFilteredItems([])
+    }, 0)
+  }
+  return (
+    <SelectPanel
+      title="Set projects"
+      renderAnchor={({children, 'aria-labelledby': ariaLabelledBy, ...anchorProps}) => (
+        <Button trailingAction={TriangleDownIcon} aria-labelledby={` ${ariaLabelledBy}`} {...anchorProps}>
+          {children ?? 'Select Labels'}
+        </Button>
+      )}
+      open={open}
+      onOpenChange={setOpen}
+      items={filteredItems}
+      selected={selected}
+      onSelectedChange={setSelected}
+      onFilterChange={onFilterChange}
+      overlayProps={{width: 'medium', height: 'large'}}
+    >
+      <SelectPanel.Message variant="noitems" title="You haven't created any projects yet">
+        <Link href="https://github.com/projects">Start your first project </Link> to organise your issues.
+      </SelectPanel.Message>
+      <SelectPanel.Message variant="nomatches" title={`No language found for `}>
```

Comment:
Ooh, interesting. Does this mean we would figure out which Message to render and ignore the rest?

Reply:
1. Actually it is similar to what you have mentioned in [your comment](https://github.com/primer/react/pull/4988/files#r1773171638) - to have the data for both state. Here is more of a variant version and yours is coming from a message object if I understand it right. Tbh, I don't like this API very much. It is cumbersome. I am just entertaining various options 😄

Reply:
2. Yeah!

Reply:
Yeah this seems great! My only suggestion is to add individual slots (children) for the title and description.